SQL: "panic: panic called with nil argument" at commit 070e54a

## tl;dr

This query causes a panic.

```
$ super -version
Version: 070e54a29

$ super -c "SELECT COALESCE (COUNT (*), (COUNT(ALL 1)));"
panic: panic called with nil argument

goroutine 1 [running]:
github.com/brimdata/super/compiler/semantic/sem.CopyExpr({0x0?, 0x0?})
	/Users/phil/work/super/compiler/semantic/sem/expr.go:404 +0xf9a
github.com/brimdata/super/compiler/semantic/sem.CopyExpr({0x10b9d600?, 0xc00013f900})
	/Users/phil/work/super/compiler/semantic/sem/expr.go:260 +0xe3f
github.com/brimdata/super/compiler/semantic/sem.copyAssignments({0xc0002259b0?, 0x10958b00?, 0xc000214f01?})
	/Users/phil/work/super/compiler/semantic/sem/op.go:522 +0xf4
github.com/brimdata/super/compiler/semantic/sem.CopyOp({0x10b9db10?, 0xc00013f950})
	/Users/phil/work/super/compiler/semantic/sem/op.go:360 +0xae
github.com/brimdata/super/compiler/semantic/sem.CopySeq(...)
	/Users/phil/work/super/compiler/semantic/sem/op.go:286
github.com/brimdata/super/compiler/semantic/sem.CopyExpr({0x10b9d900?, 0xc000225a10})
	/Users/phil/work/super/compiler/semantic/sem/expr.go:390 +0x17c6
github.com/brimdata/super/compiler/semantic/sem.copyExprs(...)
	/Users/phil/work/super/compiler/semantic/sem/expr.go:411
github.com/brimdata/super/compiler/semantic/sem.CopyExpr({0x10b9d690?, 0xc0002266c0})
	/Users/phil/work/super/compiler/semantic/sem/expr.go:287 +0x13a8
github.com/brimdata/super/compiler/semantic.(*translator).genAggregateOutput(0xc0002f4af0, {0x10b928c8, 0xc000215020}, {0xc000226740?, 0xc0002266c0?, 0xc00021efd0?}, {0x0, 0x0, 0x0}, {0xc0002267c0, ...})
	/Users/phil/work/super/compiler/semantic/sql.go:232 +0x33b
github.com/brimdata/super/compiler/semantic.(*translator).genAggregate(0xc0002f4af0, {0x0, 0x2a}, {0xc000226740?, 0x0?, 0x1}, {0x0, 0x0}, {0x0, 0x0, ...}, ...)
	/Users/phil/work/super/compiler/semantic/sql.go:206 +0x11d4
github.com/brimdata/super/compiler/semantic.(*translator).sqlSelect(0xc0002f4af0, 0xc0003886c0, {0x0?, 0x0?, 0xc000228060?})
	/Users/phil/work/super/compiler/semantic/sql.go:58 +0x86b
github.com/brimdata/super/compiler/semantic.(*translator).sqlQueryBody(0x0?, {0x10b9bbb0?, 0xc0003886c0?}, {0x0?, 0x0?, 0x0?})
	/Users/phil/work/super/compiler/semantic/sql.go:451 +0x28b
github.com/brimdata/super/compiler/semantic.(*translator).sqlQueryBody(0xc0002f4af0, {0x10b9bb80?, 0xc00013f810}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/sql.go:459 +0x1ab
github.com/brimdata/super/compiler/semantic.(*translator).semOp(0xc0002f4af0, {0x10b9c1e0?, 0xc000224f00}, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/semantic/op.go:541 +0xc29
github.com/brimdata/super/compiler/semantic.(*translator).seq(...)
	/Users/phil/work/super/compiler/semantic/op.go:33
github.com/brimdata/super/compiler/semantic.Analyze({0x10ba43e0?, 0xc0002255f0?}, 0xc000220ce0, 0xec08aa5?, 0x0)
	/Users/phil/work/super/compiler/semantic/analyzer.go:26 +0x1d0
github.com/brimdata/super/compiler.Analyze(...)
	/Users/phil/work/super/compiler/package.go:27
github.com/brimdata/super/compiler.CompileWithAST(0xc0002255f0, 0xed37b65?, 0xc0002f4a80, 0x1, 0xc, {0x0, 0x0, 0x0})
	/Users/phil/work/super/compiler/package.go:68 +0x65
github.com/brimdata/super/compiler.(*compiler).NewQuery(0x10ba4488?, 0xc00042e000?, 0xc00054d9a8?, {0x0?, 0x90?, 0x10b1f340?}, 0xec644a0?)
	/Users/phil/work/super/compiler/compiler.go:42 +0x45
github.com/brimdata/super/runtime.CompileQuery({0x10ba4488?, 0xc00042e000?}, 0xc000388750, {0x10b912a8, 0xc00008a640}, 0xc000220ce0, {0x0, 0x0, 0x0})
	/Users/phil/work/super/runtime/compiler.go:34 +0x10f
github.com/brimdata/super/cmd/super/root.(*Command).Run(0xc000422248, {0xc00003e8f0, 0x0, 0x0})
	/Users/phil/work/super/cmd/super/root/command.go:153 +0x591
github.com/brimdata/super/pkg/charm.path.run({0xc00042a028, 0x1, 0x1}, {0xc00003e8f0, 0x0, 0x0?})
	/Users/phil/work/super/pkg/charm/path.go:11 +0x7b
github.com/brimdata/super/pkg/charm.(*Spec).Exec(0x11d0e380, {0xc00003e8e0, 0x2, 0x2})
	/Users/phil/work/super/pkg/charm/charm.go:74 +0x1fa
main.main()
	/Users/phil/work/super/cmd/super/main.go:39 +0x5b
```

## Details

Repro is with super commit 070e54a, which is associated with the merge of #6388. The query above is a simplification of a query from [a sqllogictest](https://github.com/dolthub/sqllogictest/blob/a1f4aa3c1d691552a4a03d9603c45bcf93d0bb9d/test/random/expr/slt_good_101.test#L76992-L76996).

The panic did not occur at commit 1b72775 just prior to the merge of #6388.

```
$ super -version
Version: 1b727752f

$ super -c "SELECT COALESCE (COUNT (*), (COUNT(ALL 1)));"
{COALESCE:1::uint64}
```

--- a/compiler/semantic/sem/expr.go
+++ b/compiler/semantic/sem/expr.go
@@ -251,6 +251,8 @@ func NewStructuredError(n ast.Node, message string, on Expr) Expr {
 
 func CopyExpr(e Expr) Expr {
 	switch e := e.(type) {
+	case nil:
+		return nil
 	case *AggFunc:
 		return &AggFunc{
 			Node:     e.Node,

